teps/0021-results-api.md

```diff
@@ -0,0 +1,772 @@
+    minimum this includes the full TaskRun and PipelineRun status/spec, but may
+    include other data about the Runs (events that triggered it, artifacts
+    produced, post-run tasks, etc.)
+- Decouple long-term Tekton result storage away from runtime storage in etcd.
+- Give clients flexibility to embed additional metadata associated with Tekton
+  Task/PipelineRuns.
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Log exporting - we will rely on
+  [existing projects](https://landscape.cncf.io/category=logging&format=card-mode&grouping=category)
+  to handle exporting of logs to remote systems.
+- Defining a system to automatically clean up completed Task/PipelineRuns.
+- Define specific authentatication/authorization mechanisms (this will likely be
+  implementation specific).
+
+## Requirements
+
+<!--
+List the requirements for this TEP.
+-->
+
+- Results integrations should be an optional add-on on top of Tekton. Tekton
+  Pipelines should be able to operate normally without the presense of a Result
+  server.
+- Result API Server hosts should be able to store multi-tenant results and share
+  the server among multiple namespaces or clusters.
+- Service providers can choose to provide their own controller and server
+  implementations to upload Results with additional service specific metadata or
+  implementation specific validation / field formats, so long as it conforms to
+  the Results API.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Result API Spec
+
+Tekton should document a versioned gRPC API (with
+[JSON/HTTP bindings](https://cloud.google.com/endpoints/docs/grpc/transcoding))
+to ingest execution and triggering details sent from a Tekton Pipelines
+controller and Triggering Event Listeners, and to serve a query API to end
+users.
+
+The query API should accept filters expressed in some query language. The choice
+of that specific query language (K8s field selectors? CEL? GraphQL? Something
+else?) is server dependent and should not block approval of this proposal.
+
+At a minimum, the API should be able to answer common queries about TaskRuns and
+PipelineRuns, such as what (if any) predefined Task or Pipeline they were
+running, what inputs were specified and which outputs were produced, how long
+each execution took, and so on.
+
+### Reference Implementation
+
+#### API Server
+
+Tekton should provide a reasonably performant and scalable API service
+implementation that runs on Kubernetes (using a replicated K8s Service), so that
+users wishing to run their own Results server can also benefit from the API.
```

Comment:
Would the plan be to use MySQL for now? (like https://github.com/tektoncd/experimental/tree/master/results ?)